_data-prepper/pipelines/configuration/processors/csv.md

```diff
@@ -23,9 +23,78 @@ delete_header | No | Boolean | If specified, the event header (`column_names_sou
 column_names | No | List | User-specified names for the CSV columns. Default value is `[column1, column2, ..., columnN]` if there are no columns of data in the CSV record and `column_names_source_key` is not defined. If `column_names_source_key` is defined, the header in `column_names_source_key` generates the event fields. If too few columns are specified in this field, the remaining column names are automatically generated. If too many column names are specified in this field, the CSV processor omits the extra column names.
 
-<!---## Configuration
+## Usage
+
+Add the following examples to your `pipelines.yaml` file, depending on how you your CSV columns are formatted.
+
+### User-specified column names
+
+The following example `pipelines.yaml` configuration points to a file named `ingest.csv` as the source. Then, the `csv` processor parses the data from the `.csv` file using the column names specified in the `column_names` setting, as shown in the following example:
+
+```yaml
+csv-pipeline:
+  source:
+    file:
+      path: "/full/path/to/ingest.csv"
+      record_type: "event"
+  processor:
+    - csv:
+        column_names: ["col1", "col2"]
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+
+When run, the processor will parse the message. Although only two column names are specified in processor settings, a third column name is autogenerated since the data contained in `ingest.csv` included three columns, `1,2,3`:
+
+```
+{"message": "1,2,3", "col1": "1", "col2": "2", "column3": "3"}
+```
+### Auto Detect column names
+The following configuration auto detects the header of a CSV file ingested through an [`s3 source`]({{site.url}}{{site.baseurl}}//data-prepper/pipelines/configuration/sources/s3/):
 
-Content will be added to this section.--->
+```yaml
+csv-s3-pipeline:
+  source:
+    s3:
+      notification_type: "sqs"
+      codec:
+        newline:
+          skip_lines: 1
+          header_destination: "header"
+      compression: none
+      sqs:
+        queue_url: "https://sqs.<region>.amazonaws.com/<account id>/<queue name>"
+      aws:
+        region: "<region>"
+  processor:
+    - csv:
+        column_names_source_key: "header"
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+
+For example, if the `ingest.csv` in the S3 bucket that the SQS queue is attached to contains the following data:
+
+```
+Should,skip,this,line
+a,b,c
+1,2,3
+```
+
+The `csv` processor will take the following event:
+
+```json
+{"header": "a,b,c", "message": "1,2,3"}
+```
+
+Then, the processor parses the event into the following output. Because `delete_header` is `true`, the header `a,b,c` is deleted from the output:
```

Comment:
Where is `delete_header` set? I don't see it in the example configuration.

Reply:
Yeah this line is kind of misleading. It's true by default. It should probably say something like "Because `delete_header` is `true` by default"